### PR TITLE
fix #7965 feat(nimbus): Prepare backend to support team

### LINKED ISSUE #7965: Team filtering backend
Users can select ‘projects’ (or team if you wanna call it that), Projects are already supported on the backend, we need to add this field in NimbusExperimenter model to support the selection of teams/projects on the backend side.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2936)

### PR DESCRIPTION
**Pull request by yashika-khurana**
Because

* We want to allow users to select the Project team and link to experiments

This commit

* Add the capability to support choosing projects option in v5 Api- get experiment by slug and get nimbus config

screenshot of `an experiment by slug api`

image

screenshot of `nimbus config api`

image

fix #7965

### CODE REVIEW

File: app/experimenter/experiments/tests/test_models.py [hunk 1]
```diff
@@ -560,6 +562,27 @@ def test_targeting_with_languages_mobile(self):
         )
         JEXLParser().parse(experiment.targeting)
 
+    def test_with_projects(self):
```
Comment (eliserichards) [inline app/experimenter/experiments/tests/test_models.py:565]: Nit: 
```suggestion
    def test_targeting_with_projects(self):
```

File: app/experimenter/nimbus-ui/src/lib/mocks.tsx [hunk 2]
```diff
@@ -393,6 +393,20 @@ export const MOCK_CONFIG: getConfig_nimbusConfig = {
       id: 3,
     },
   ],
+  projects: [
+    {
+      name: "Pockets",
```
Comment (eliserichards) [inline app/experimenter/nimbus-ui/src/lib/mocks.tsx:398]: Nit: 
```suggestion
      name: "Pocket",
```

File: app/experimenter/nimbus-ui/src/lib/mocks.tsx [hunk 3]
```diff
@@ -579,6 +593,7 @@ export const MOCK_EXPERIMENT: Partial<getExperiment["experimentBySlug"]> = {
   locales: [{ name: "Quebecois", id: 1 }],
   countries: [{ name: "Canada", id: 1 }],
   languages: [{ name: "English", id: 1 }],
+  projects: [{ name: "Pockets", id: 1 }],
```
Comment (eliserichards) [inline app/experimenter/nimbus-ui/src/lib/mocks.tsx:596]:
```suggestion
  projects: [{ name: "Pocket", id: 1 }],
```

File: app/experimenter/nimbus-ui/schema.graphql [hunk 4]
```diff
@@ -549,6 +551,13 @@ type NimbusOutcomeType {
   metrics: [NimbusOutcomeMetricType]
 }
 
+type NimbusProjectType {
+  id: Int
+  name: String
+  slug: String
+  nimbusexperimentSet: [NimbusExperimentType!]!
```
Comment (eliserichards) [inline app/experimenter/nimbus-ui/schema.graphql:558]: This is probably a stupid question, but why do we need anything more than `id` and `name` here? (or `name` and `slug`, I'm not sure the difference between slug and id)

**Review by eliserichards — CHANGES_REQUESTED** [review]
This is excellent! The tests are super thorough 🎉 Just a few nits and a question--

File: app/experimenter/nimbus-ui/schema.graphql (hunk 4 above)
Comment (yashika-khurana) [reply app/experimenter/nimbus-ui/schema.graphql:558]: so the thing is Projects is the existing model, but here are some insights `Slug is a URL friendly short label for specific content. It only contain Letters, Numbers, Underscores or Hyphens. Slugs are commonly save with the respective content and it pass as a URL string.` whereas id is django generated unique keys, so we have name user friendly and slug as identifier

**Review by eliserichards — APPROVED** [review]
Lovely! 💯

**Pull request merged**